### PR TITLE
[tvOS, visionOS, watchOS] Skip install step for WebKit process extensions

### PR DESCRIPTION
**Pull request by pvollan**
#### 42bebeed39354320e700d7343e42f2e31f172a74
<pre>
[tvOS, visionOS, watchOS] Skip install step for WebKit process extensions
<a href="https://bugs.webkit.org/show_bug.cgi?id=276210">https://bugs.webkit.org/show_bug.cgi?id=276210</a>
<a href="https://rdar.apple.com/130775557">rdar://130775557</a>

Reviewed by Alexey Proskuryakov.

Skip install step for WebKit process extensions on platforms where they are not enabled.
We need to explicitly skip the install step for visionOS and watchOS, since they
otherwise will match an iPhone SDK predicate.

* Source/WebKit/Configurations/BaseExtension.xcconfig:

Canonical link: <a href="https://commits.webkit.org/280757@main">https://commits.webkit.org/280757@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/92a4d1a6367304be10f7ada97c47bc67776ea4ce

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57497 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36825 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9972 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61119 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7942 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44458 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8130 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/46561 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5631 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59527 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49667 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27424 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/31344 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6945 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/53309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7242 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62798 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1410 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/7334 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/53819 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1417 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49692 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/53914 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1205 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8585 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32654 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33739 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34824 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33485 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/Configurations/BaseExtension.xcconfig [hunk 1]
```diff
@@ -29,8 +29,11 @@ CLANG_INSTRUMENT_FOR_OPTIMIZATION_PROFILING = NO;
 INSTALL_PATH = $(INSTALL_PATH_PREFIX)/System/Library/ExtensionKit/Extensions;
 INSTALL_PATH[sdk=*simulator] = $(WK_INSTALL_PATH_PREFIX)/System/Library/ExtensionKit/Extensions;
 INSTALL_PATH[sdk=iphone*17.4*] = $(WK_INSTALL_PATH_PREFIX)/System/Library/ExtensionKit/Extensions;
-SKIP_INSTALL = YES;
-SKIP_INSTALL[sdk=iphone*] = NO;
```
Comment (aproskuryakov) [inline Source/WebKit/Configurations/BaseExtension.xcconfig:33]: Isn't the fix to switch from `iphone` to `ios` here?

Comment (pvollan) [reply Source/WebKit/Configurations/BaseExtension.xcconfig:33]: Good point, will fix!

Thanks for reviewing!

Comment (aproskuryakov) [reply Source/WebKit/Configurations/BaseExtension.xcconfig:33]: Wait no, `ios` does not exist. I'm not sure if there is even a better way? @emw-apple would know better, but r+ in the meanwhile to unbreak things.

**Review by aproskuryakov — APPROVED** [review]

File: Source/WebKit/Configurations/BaseExtension.xcconfig [hunk 2]
```diff
@@ -30,7 +30,7 @@ INSTALL_PATH = $(INSTALL_PATH_PREFIX)/System/Library/ExtensionKit/Extensions;
 INSTALL_PATH[sdk=*simulator] = $(WK_INSTALL_PATH_PREFIX)/System/Library/ExtensionKit/Extensions;
 INSTALL_PATH[sdk=iphone*17.4*] = $(WK_INSTALL_PATH_PREFIX)/System/Library/ExtensionKit/Extensions;
 SKIP_INSTALL = YES;
-SKIP_INSTALL[sdk=iphone*] = NO;
+SKIP_INSTALL[sdk=ios*] = NO;
```
Comment (aproskuryakov) [inline Source/WebKit/Configurations/BaseExtension.xcconfig]: To be clear, this actually doesn't work AFAIK, my r+ is for the previous iteration.

Comment (pvollan) [reply Source/WebKit/Configurations/BaseExtension.xcconfig]: Sounds good! I have gone back to the previous iteration in the latest patch.

Thanks for approving!

**Pull request merged**